Lib/asyncio/locks.py

```diff
@@ -346,9 +346,8 @@ class Semaphore(_ContextManagerMixin, mixins._LoopBoundMixin):
     def __init__(self, value=1):
         if value < 0:
             raise ValueError("Semaphore initial value must be >= 0")
+        self._waiters = None
```

Comment:
> at the cost of making the code less like `Lock`

That's exactly I'm trying to avoid. Are you sure you not want to update this for `Lock` and `Semaphore` together, in a separate commit?

Reply:
Well, the original Lock code initialized the waiters queue lazily, and this makes sense to me -- Locks are supposed to be lightweight, and if you're just a little bit lucky a Lock may never be contended, hence saving the allocation and deallocation of an empty deque. OTOH the original Semaphore code always created a deque, and that makes sense to me too -- it's not as heavily used as Lock, so optimization vs. clean code is weighed differently, and I expect that the use cases for Semaphores (other than Semaphore(1) as a Lock) tend to be such that they will need the deque anyway.

But in the end I'm fine with your choice here.